client-cm/src/main/java/com/sequenceiq/cloudbreak/cm/client/tracing/StackTraceBasedCmApiNameExtractor.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.cloudbreak.cm.client.tracing;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StackTraceBasedCmApiNameExtractor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackTraceBasedCmApiNameExtractor.class);
+
+    public Optional<String> getCmApiName(StackTraceElement[] stackTrace) {
```

Comment:
what's the performance impact of this? We make a lot of call to CM, if we inspect every calls stacktrace, could it increase the load significantly?

Reply:
Let me check. My guess is not much.

Reply:
I will rewrite to use StackWalker

https://stackoverflow.com/a/49091729